draft-ietf-tls-esni.md

```diff
@@ -1638,6 +1640,30 @@ ClientHelloInner and authenticating all inputs to the ClientHelloInner
 encrypted and authenticated the "server_name" extension, which left the overall
 ClientHello vulnerable to an analogue of this attack.
 
+### ClientHelloInner Packet Amplification Mitigation
+
+Client-facing servers must decompress EncodedClientHelloInners. A malicious
+attacker may craft a packet which takes excessive resources to decompress
+or may be much larger than the incoming packet:
+
+* If looking up a ClientHelloOuter extension takes time linear in the number of
+  extensions, the overall decoding process would take O(M\*N) time, where
+  M is the number of extensions in ClientHelloOuter and N is the
+  size of OuterExtensions.
+
+* If OuterExtensions contains duplicate extensions, an attacker could cause the
```

Comment:
Nit: Given the weird ambiguity around what a "duplicate extension" is, how about "If the same ClientHelloOuter extension can be copied multiple times".